### ISSUE TITLE
Continuous Control with Deep Reinforcement Learning

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#21]
# Key Idea

- Extend DQN to ***continuous action space*** by adding ***actor-critic network***
- Stabilizing learning through ***soft target updates***
- Introduce ***Batch Normalization*** to RL domain
- Construct exploration policy by adding noise

# Previous Approaches

### Deep Q Network

- Approximate action-value function(Q-function) using deep neural network
- Use ***replay buffers*** to mimic iid sampling of episodes
- Employ ***target network*** to stabilize convergence
- Limited to finite, discrete action space since it searches action that minimizes Q

# Formulation
![image](https://user-images.githubusercontent.com/46308407/104118581-928a2600-536d-11eb-9443-e359b956984e.png)

- Actor network learns policy using chain-rule based back-propagation
 
![image](https://user-images.githubusercontent.com/46308407/104118584-99b13400-536d-11eb-98fe-5a9388168a8f.png)

- Critic network learns Q function as in DQN - minimizing MSE loss w.r.t. target

![image](https://user-images.githubusercontent.com/46308407/104118586-9fa71500-536d-11eb-901b-dd132bd5eab3.png)

- Adopts ***replay buffers*** from DQN to sample uncorrelated episodes
- Use ***target network*** from DQN while updating ***softly*** by linear interpolation

![image](https://user-images.githubusercontent.com/46308407/104118592-acc40400-536d-11eb-89db-3149fcb974f1.png)

- Use Batch Normalization for scaling and apply noise for exploration policy

![image](https://user-images.githubusercontent.com/46308407/104118598-b3527b80-536d-11eb-941b-21ac5d2f741b.png)

SUDO algorithm for DDPG

![image](https://user-images.githubusercontent.com/46308407/104118606-bb122000-536d-11eb-803c-2dddb31ca85a.png)

# Result

- Quicker to converge compared to DQN
- Able to learn on continuous action space
